src/shared/services/addresses/addresses.service.ts

```diff
@@ -15,9 +16,40 @@ export class AddressesService {
 
   getAddressesForPostcode(postcode: string): Observable<Array<AddressModel>> {
     return this.http
-      .get(this.appConfig.getPostcodeLookupUrl().replace('${postcode}', postcode))
+      .get(this.appConfig.getPostcodeLookupUrl()
+        .replace('${postcode}', postcode))
       .pipe(
-        map(response => response.json())
-      );
+        map(res=> res.json().results)
+       ).pipe(
+        map(output => output.map(addresses=>
+            AddressesService.mapToAddressModel(addresses["DPA"]))
+      ));
   }
+
+  static mapToAddressModel(address:any) {
+    const addressModel = new AddressModel();
+    let addressLine = AddressesService.tidyEmptyValues(
+      `${address.ORGANISATION_NAME} ${address.DEPARTMENT_NAME} ${address.PO_BOX_NUMBER}`)
+    if (addressLine != '') addressModel.AddressLine1 = addressLine;
+    addressModel.AddressLine2 = AddressesService.tidyEmptyValues(
+      `${address.BUILDING_NAME} ${address.SUB_BUILDING_NAME} ${address.BUILDING_NUMBER} ${address.THOROUGHFARE_NAME}`);
+    addressModel.AddressLine3 = AddressesService.tidyEmptyValues(
+      `${address.DEPENDENT_LOCALITY} ${address.DOUBLE_DEPENDENT_LOCALITY} ${address.DEPENDENT_THOROUGHFARE_NAME}`);
+    addressModel.PostCode = address.POSTCODE;
+    addressModel.PostTown = address.POST_TOWN;
+    addressModel.Country = 'UK';
```

Comment:
can you introduce a constant for 'UK' please

Reply:
👍